### ISSUE TITLE
Callback context

### ISSUE THREAD

**Issue by subzey** [ai/nanoevents#10]
https://github.com/ai/nanoevents/blob/8b91c1120368b05c5dccd4e8b2148fc49428f820/index.js#L84

Previously, when it was a `for` loop, `this` was referring to the class instance itself. Now, within `.map` function it refers to _global_. I'm not really sure it's useful.

We could:
- Bring back the old behavior: `.map(function(){ /*...*/ }, this)`, or
- Leave as is (same as `i.call(null, args)`)

**Comment by ai** [comment]
So, what is an issue? It was a major release, we change API.

**Comment by ai** [comment]
Honestly, I thought that it was `global` in previous API :D.

Anyway, in public API `this` is not documented. I think that if you want to use `this` you should `bind()` callback to explicit context:

```js
on((function () {
  this.method()
}).bind(app))
```

Or use `=>` functions.

**Comment by subzey** [comment]
Yup, a major release can break the API, especially it's undocumented parts, you're right.

It's just now this line looks weird. Imagine a developer that wants to know, with which context the handler would be called (maybe intending to write a `.d.ts` type definition). There's no trace of that in the spec. They opens up the code - ah! `this`! But it's not the instance, it's not a `null` (one would expect that when a `use strict`'ed function is called), it's always a `global`.

To sum things up: `i.call(this, args)` could be changed into `i.call(null, args)` with no filesize penalty. But this would make the behavior crystal clear and more compatible with strict mode.

Anyways, it's rather a line comment than a "real" issue, feel free to close it. _(I don't know how to comment a line on GH without a new issue, sorry!)_

**Comment by ai** [comment]
Most of JS event listeners have `global` in the default context. Using `null` will not be expected.

Maybe we can put comment `// this === global` to sources? And add note to JSDoc?

**Comment by subzey** [comment]
Are you sure it's `global` and not the `undefined` that becomes `global` if the function is in loose mode (i.e. not strict one)?

Btw, just checked node internal `EventEmitter`: event handler's context is the EventEmitter instance.
Same is for DOM events: [`this` is the DOM element on which addEventListener was called](https://www.w3.org/TR/dom/#dispatching-events).

**Comment by ai** [comment]
Thanks, my mistake. Yeap EventEmitter sends emitter and DOM sends DOM node.

But right now I think comment and docs is still better idea:
1. Changing to `null` will be the major change.
2. Will will make the major release without a good reason for it. Comment fix problem with readability.

**Comment by subzey** [comment]
_Assuming "We will **not** make the major..."_

Looks good to me, thanks! The one thing I didn't get though: is it you who will add a comment, or should I issue a PR?

**Comment by ai** [comment]
Yeap, send PR with comment :)
